app/src/App/App.tsx

```diff
@@ -1,18 +1,32 @@
-import {BrowserRouter as Router, Route, Routes} from 'react-router-dom';
+import {BrowserRouter as Router, Route, Routes, useLocation} from 'react-router-dom';
 import {HomeLayout} from '../Layout/Layout';
 import {HomePage} from '../Homepage/Homepage';
 import {ClassList} from '../Directories/Directory';
 import {LoginPage} from '../Login/LoginPage';
 import {SignupPage} from '../Login/SignupPage';
 import {CoursePage} from '../CoursePages/CoursePage';
 import {ReviewPage} from '../Rating/Rating';
+import {useEffect} from 'react';
+
+// Utility component, scrolls to the top of the 
+// page upon route change
+function ScrollToTop() {
+  const { pathname } = useLocation();
```

Comment:
Is the curly braces needed? Or can you just use pathname without it?

Reply:
{} extracts the pathname property from useLocation. In practice, it makes it so the scroll only resets when the URL changes, not when a navigation button is clicked. 

Should the scroll reset when 'CSE 300s' is clicked when you're already on the CSE 300s page?